spec/controllers/comments_controller_spec.rb

```diff
@@ -854,6 +1079,25 @@
               "Comment thread successfully frozen!"
             )
           end
+
+          context "when comment is a guest reply to user who turns off guest replies afterwards" do
+            let(:reply) do
+              reply = create(:comment, :by_guest, commentable: comment)
+              comment.user.preference.update!(guest_replies_off: true)
+              reply
+            end
+
+            it "freezes reply and redirects with success message" do
+              fake_login_admin(admin)
+              put :freeze, params: { id: reply.id }
```

Comment:
Where is this implemented ?

Reply:
It's checking that the validation on the model only runs on :create. If it always runs, freezing the guest reply after the user turns off guest replies will fail validation.

I could not figure out how do this as a model test, would appreciate pointers there if that is preferred. I was hung up on how to do the freezing. Just updating the `iced` attribute didn't seem correct.

Reply:
I am trying to understand the test and reconcile it with the code I can see.

I think the test says a comment which is from a guest becomes a frozen thread if the user of a parent comment disables commenting by guests. 

I don't think that is intended so I suspect I am missing something.

Reply:
User makes comment. Guest replies. User turns off "allow guest replies". Admin tries to freeze guest reply.

Intended: Guest reply is frozen.
Possible problem this test would catch: Freezing fails because the validation of guest reply says that guest replies aren't allowed for the user.

Reply:
@zz9pzza It's basically trying to make sure we don't run into a problem like [AO3-6373](https://otwarchive.atlassian.net/browse/AO3-6373).